clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/common/AzureUtilities.groovy

```diff
@@ -112,6 +103,37 @@ class AzureUtilities {
     resourceId.split(PATH_SEPARATOR).last()
   }
 
+  // For id = "/subscriptions/***-***-***/resourceGroups/***/providers/Microsoft.Network/networkInterfaces/nic1/ipConfigurations/ipconfig1"
+  //   this method return "nic1"
+  static String getResourceNameFromId(String id) {
+    if (id == null) {
+      return null
+    }
+
+    def vals = id.split(PATH_SEPARATOR)
+
+    if (vals.length > 8) {
+      return vals[8] // see vals.findIndexOf { it == "Microsoft.Network"} + 2
```

Comment:
The implementation for the method is correct because the id starts with '/' therefore the value of first element in the split array is "".

It's an overkill to add parameters for each call with the expected "provider"; in fact before we call the method we don't even know who that provider is because that's information only available within the ResourceId.

There's a comment in there that should make it clearer in case the index into the array changes somehow. We could promote that index to a constant within the AzureUtilities if you think that will make it more clear (will have to add two though, one for the resource name and one for the type). At this time I'm more one the side of not to change the current implementation.